Function documention should not mention what the default value is in the parameter description #24

xtian0329 opened the pull request:
- remove the default values from function documentations
- keep the description of the variables with NULL default value 
note: some default values that are **not** NULL remain because there are additional explanations for user to understand the meaning behind the value, let me know if this is recommended to be modified.

xtian0329 committed:
Function documention should not mention what the default value is in the parameter description #24:
- remove the default values from function documentations
- keep the description of the variables with NULL default value
note: some default values that are not NULL remain because there are additional explanations for user to understand the meaning behind the value.

ofarrelle reviewed (changes requested):
Thank you for this work!
Just a few more I spotted that need changing, maybe do a Ctrl-F for "default"

xtian0329 commented:
Hey Ethan, yeah, I kept some non-NULL default values that have some explanations in the function description, do you mean that those ones should be excluded as well?
Ex. "@param contrast_number Integer. Index specifying which contrast to extract results from. Defaults to \code{1} corresponding to the first contrast.", or I should paraphrase it a bit for example, "@param contrast_number Integer. Index specifying which contrast to extract results from. The value of \code{1} corresponding to the first contrast."

ofarrelle commented:
Yes I still think this is a good change to make, mostly for simplicity in the future - if we decide to change a function's default value, ideally we only need to change it in one place: the function argument, and not the documentation. If the user is curious what the default is, they can still see it in the help page.

Your proposal is good too, because it tells the user how to _interpret_ the argument, and provides an example, instead of just stating what the default value is. If we change the default value, we don't need to change the example.

I personally think the documentation for `contrast_number` is clear enough and we don't need to keep the example or the default value.

xtian0329 committed:
Function documention should not mention what the default value is in the parameter description #24:
- removed all default
- adjust the wording for arguments which necessitate the explanation of the default value
Note: I also remove the sentence that contains default is set to NULL since it is already in the function signature and for our cases we don't have the scenario where a new default will be computed by the function.

ofarrelle reviewed (approved):
Looks good!
Thank you for all the hard work on this project :)
